graphql/schemabuilder/pagination.go

```diff
@@ -420,11 +448,36 @@ func (sb *schemaBuilder) buildPaginatedField(typ reflect.Type, f interface{}) (*
 	return ret, nil
 }
 
-func (funcCtx *funcContext) extractPaginatedRetAndErr(nodeKey string, out []reflect.Value, args interface{}, retType graphql.Type) (interface{}, error) {
+func getEmbedIndex(typ reflect.Type) int {
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		if field.Type == reflect.TypeOf(PaginationArgs{}) {
+			return i
+		}
+	}
+	return 0
+}
+
+func (funcCtx *funcContext) extractPaginatedRetAndErr(nodeKey string, out []reflect.Value, args interface{}, retType graphql.Type, embedsArgs bool) (interface{}, error) {
 	var result interface{}
-	connectionArgs, _ := args.(ConnectionArgs)
+	var paginationArgs PaginationArgs
+
+	// If the pagination args are not embedded then they need to be extracted out of ConnectionArgs
+	// struct and setup for the slicing functions.
+	if !embedsArgs {
+		connectionArgs, _ := args.(ConnectionArgs)
+		paginationArgs = PaginationArgs{
+			First:  connectionArgs.First,
+			Last:   connectionArgs.Last,
+			After:  connectionArgs.After,
+			Before: connectionArgs.Before,
+		}
+	} else {
+		fieldInd := getEmbedIndex(reflect.TypeOf(args))
+		paginationArgs, _ = reflect.ValueOf(args).Field(fieldInd).Interface().(PaginationArgs)
+	}
 
-	result, err := getConnection(nodeKey, castSlice(out[0].Interface()), connectionArgs)
+	result, err := getConnection(nodeKey, castSlice(out[0].Interface()), paginationArgs)
```

Comment:
interesting. i think if we are passing down the pagination args, we should assume that the underlying resolver has handled passing back the exact correct results